core/src/stdlib/pyscript/events.py

```diff
@@ -84,84 +84,135 @@ def remove_listener(self, *listeners):
             self._listeners = []
 
 
-def when(target, *args, **kwargs):
+def when(event_type, selector=None):
     """
-    Add an event listener to the target element(s) for the specified event type.
+    A decorator to handle DOM events or custom Event objects.
 
-    The target can be a string representing the event type, or an Event object.
-    If the target is an Event object, the event listener will be added to that
-    object. If the target is a string, the event listener will be added to the
-    element(s) that match the (second) selector argument.
+    For DOM events, specify the `event_type` (e.g. `"click"`) and a `selector`
+    for target elements. For custom `Event` objects, just pass the `Event`
+    instance as the `event_type`. It's also possible to pass a list of `Event`
+    objects. The `selector` is required only for DOM events. It should be a
+    CSS selector string, Element, ElementCollection, or list of DOM elements.
 
-    If a (third) handler argument is provided, it will be called when the event
-    is triggered; thus allowing this to be used as both a function and a
-    decorator.
+    The decorated function can be either a regular function or an async
+    function. If the function accepts an argument, it will receive the event
+    object (for DOM events) or the Event's result (for custom events). A
+    function does not need to accept any arguments if it doesn't require them.
+
+    ```python
+    from pyscript import when, display
+
+    # Handle DOM events.
+    @when("click", "#my-button")
+    def handle_click(event):
+        display("Button clicked!")
+
+    # Handle custom events.
+    my_event = Event()
```

Comment:
IWBNI you also explained how to get Event() to be an actual event like "click", "#my-button"